src/extra/utils/functions.py

```diff
@@ -23,6 +23,53 @@ def gaussian(x, y0, A, μ, σ):
     return y0 + (A / (σ * np.sqrt(2*np.pi))) * np.exp(-(x - μ)**2 / (2 * σ**2))
 
 
+def fit_gaussian(ydata, return_all=False, xdata=None, p0=None, **kwargs):
```

Comment:
Signature nitpicking :nerd_face: :

`return_all` seems confusingly similar to the `full_output` parameter in `curve_fit()`, but you'd have to specify both `return_all=True, full_output=True` to get all the possible output. Can we make do with just one parameter, preferably following scipy's name?

It also seeems odd to me that this comes between ydata & xdata. I think passing booleans as positional arguments usually makes code less clear, unless there's a really obvious meaning:

```python
fit_info = fit_gaussian(arr, True)   # What does True mean?

fit_info = fit_gaussian(arr, full_output=True)  # Clearer
```

So I might make the option keyword-only.

Reply:
Yes, both of those make sense :+1:

Reply:
Fixed in bc05cbf.